nats/src/main.rs

```diff
@@ -25,8 +25,11 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     // returns when provider receives a shutdown control message
     let host_data = load_host_data()?;
     let provider = if let Some(c) = host_data.config_json.as_ref() {
-        let config: ConnectionConfig = serde_json::from_str(c)
-            .expect("JSON deserialization from connection config should have worked");
+        // sanitize in case we receive escaped JSON with literal quotes and backslashes
+        let c = unescape::unescape(c).unwrap(); // we want to panic here if we can't do this
```

Comment:
Shouldn't we just return the error instead? That should exit the process rather than full panicking